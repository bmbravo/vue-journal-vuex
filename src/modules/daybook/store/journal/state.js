export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: 'Magna aliqua aliqua sunt elit sunt cupidatat sunt eiusmod elit. Duis ad nisi culpa nostrud consequat aute irure. Sit Lorem incididunt sunt magna Lorem cillum ut consequat minim enim adipisicing deserunt tempor velit. Culpa sit non enim et duis culpa fugiat esse sunt reprehenderit laboris eu dolor sunt.',
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: 'Dolore est quis cillum aute do adipisicing laborum. Ut esse proident excepteur reprehenderit velit anim laborum officia enim commodo. Dolor velit pariatur adipisicing tempor ea deserunt veniam. Ex sint irure pariatur aliquip velit commodo mollit est magna enim. Id sint nostrud excepteur culpa minim laboris excepteur consequat adipisicing tempor amet. Esse mollit dolore enim ipsum tempor excepteur culpa id. Aliquip aute ullamco incididunt aliqua voluptate nostrud proident.',
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: 'Nostrud tempor Lorem qui duis cupidatat veniam elit ex do enim. Sit fugiat dolor sunt ullamco non tempor cillum labore magna fugiat. Ut non ad est culpa cupidatat esse ut et reprehenderit excepteur.',
      picture: null,
    },
  ],
});
