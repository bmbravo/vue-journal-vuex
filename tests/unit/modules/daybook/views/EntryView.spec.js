import { createStore } from 'vuex';
import { shallowMount } from '@vue/test-utils';
import journal from '@/modules/daybook/store/journal';
import { journalState } from '../../../mock-data/test-journal-state';
import EntryView from '@/modules/daybook/views/EntryView';
import Swal from 'sweetalert2';

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

jest.mock('sweetalert2', () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn(),
}));

describe('Pruebas en el EntryView', () => {
  const store = createVuexStore(journalState);

  store.dispatch = jest.fn();

  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryView, {
      props: {
        id: 'ABC123',
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test('Debe de sacar al usuario porque el Id no existe', () => {
    const wrapper = shallowMount(EntryView, {
      props: {
        id: 'este id no existe en el store',
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });

    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
  });

  test('Debe de mostrar la entrada correctamente', () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(mockRouter.push).not.toHaveBeenCalled();
  });

  test('Debe de borrar la entrada y salir', (done) => {
    Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }));

    wrapper.find('.btn-danger').trigger('click');

    expect(Swal.fire).toHaveBeenCalledWith({
      title: '¿Está seguro?',
      text: 'Una vez borrado, no se podrá recuperar la entrada',
      showDenyButton: true,
      confirmButtonText: 'Si, estoy seguro',
    });

    setTimeout(() => {
      expect(store.dispatch).toHaveBeenCalledWith(
        'journal/deleteEntry',
        'ABC123'
      );
      expect(mockRouter.push).toHaveBeenCalled();
      done();
    }, 1);
  });
});
