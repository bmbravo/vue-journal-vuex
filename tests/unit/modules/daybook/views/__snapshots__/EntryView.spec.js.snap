// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView Debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">10</span><span class="mx-1 fs-3">Julio</span><span class="mx-2 fs-4 fw-light">2023, Lunes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedio hoy?"></textarea></div>
<img src="https://res.cloudinary.com/ddqeyulq2/image/upload/v1687838261/gsfgmokyunylybm5mcbz.png" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
